skbio/sequence/_sequence.py

```diff
@@ -886,17 +880,17 @@ def __getitem__(self, indexable):
         seq = self._bytes[indexable]
         positional_metadata = self._slice_positional_metadata(indexable)
 
-        return self._to(sequence=seq, positional_metadata=positional_metadata)
+        return self._constructor(
+            sequence=seq,
+            metadata=self.metadata,
```

Comment:
`Sequence._constructor` doesn't propagate, it mainly exists over `__class__` to allow subclasses to suppress validation. `TabularMSA._constructor_` (note it is a *sunder*) propagates but doesn't exist on `Sequence`.